pupil_src/shared_modules/annotations.py

```diff
@@ -10,220 +10,297 @@
 """
 
 import csv
-
-# logging
 import logging
 import os
-from itertools import chain
+from abc import ABCMeta, abstractmethod
+from collections import namedtuple
 
-import numpy as np
-from OpenGL.GL import *
 from pyglui import ui
-from pyglui.cygl.utils import RGBA, draw_polyline
-from pyglui.pyfontstash import fontstash
-from pyglui.ui import get_opensans_font_path
 
-import player_methods as pm
 import file_methods as fm
-from glfw import glfwGetCurrentContext, glfwGetWindowSize
+import player_methods as pm
+import zmq_tools
 from plugin import Analysis_Plugin_Base, Plugin
 
 logger = logging.getLogger(__name__)
 
 
-class Annotation_Capture(Plugin):
-    """Describe your plugin here
+def create_annotation(label, timestamp, duration=0.0, **custom_fields):
+    """
+    Returns a dictionary in the format needed to send annotations
+    to an annotation plugin via the ICP.
+    
+    See python/remote_annotations.py in pupil-helpers for an example.
+
+    :param custom_fields:
+    """
+    return {
+        "topic": "annotation",
+        "label": label,
+        "timestamp": timestamp,
+        "duration": duration,
+        **custom_fields,
+    }
+
+
+AnnotationDefinition = namedtuple("AnnotationDefinition", "label hotkey")
+
+
+class AnnotationPlugin(Plugin, metaclass=ABCMeta):
+    """
+    Base for player and capture plugins that support adding and removing
+    annotations and the corresponding quickbar buttons
     """
 
-    icon_chr = chr(0xE866)
+    # class _AnnotationUI:
+    #     # no namedtuple because we cannot set the menu button at creation
+    #     # (this button references the _AnnotationInfo for deletion)
+    #     def __init__(self, definition, button_quickbar=None, button_menu=None):
+    #         self.definition = definition
+    #         self.button_quickbar = button_quickbar
+    #         self.button_menu = button_menu
+
+    _AnnotationButtons = namedtuple("_AnnotationButtons", "quickbar menu")
+
+    icon_chr = chr(0xe866)
     icon_font = "pupil_icons"
 
-    def __init__(self, g_pool, annotation_definitions=(("My annotation", "E"),)):
+    def __init__(self, g_pool, annotation_definitions=None):
```

Comment:
What is the reasoning behind this change?

Reply:
This is the suggested way to add list default arguments, right?